zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java

```diff
@@ -1142,6 +1148,7 @@ public Builder setInputVector(String inputVector) {
 
         public Builder setCweId(int cweId) {
             this.cweId = cweId;
+            this.addCweTag(cweId);
```

Comment:
- This should be done when the alert is being built. There should be a test showing that setting multiple IDs does not lead to multiple tags.
- This behaviour should be documented (in `build`?).